### PR TITLE
migrate project to uv package/project manager

### PR DESCRIPTION
**Pull request by jfchevrette**
Based on WIP https://github.com/app-sre/python-project-template/pull/3

* Base image now `ubi9/python-312` instead of `ubi9/ubi-minimal` + install python/pip ourselves
* New Dockerfile stage `builder` which installs uv and builds the project. 
* `test` stage now uses `builder` stage
* Makefile `test` target now runs uv locally
* All ruff exceptions have been carried over and were not in scope to be fixed as part of this

Tested by generating a bundle before changes and after changes and comparing sha256 checksum

```sh
# Run from qontract-server
$ make bundle && sha256sum bundle/*
...
04aefb834df57f4a20aeac98ef9bc9448e2b950508a57666efe0ef118bb4f90b  bundle/bundle.json
04aefb834df57f4a20aeac98ef9bc9448e2b950508a57666efe0ef118bb4f90b  bundle/bundle.json.before
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,18 +1,63 @@
...
+]
+
+[project.urls]
+homepage = "https://github.com/app-sre/qontract-validator"
+repository = "https://github.com/app-sre/qontract-validator"
+documentation = "https://github.com/app-sre/qontract-validator"
+
+[dependency-groups]
+dev = [
+    "ruff ~=0.7.3",
```
Comment (chassing) [inline pyproject.toml:25]:
```suggestion
    "ruff ~=0.7",
```

File: pyproject.toml [hunk 2]
```diff
@@ -1,18 +1,63 @@
...
+    "mypy ~=1.13",
+    "pytest ~=8.2.0",
```
Comment (chassing) [inline pyproject.toml:27]:
```suggestion
    "pytest ~=8.2",
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,18 +1,63 @@
...
+    "pytest-cov ~=5.0.0",
```
Comment (chassing) [inline pyproject.toml:28]:
```suggestion
    "pytest-cov ~=6.0",
```

File: pyproject.toml [hunk 4]
```diff
@@ -26,15 +71,45 @@ extend-select = [
     "UP",
 ]
 ignore = [
```
Comment (chassing) [inline pyproject.toml:73]: Please also add a `select = ["ALL"]`

**Review by chassing — APPROVED** [review]
Just some minor nits

File: pyproject.toml (hunk 4 above)
Comment (jfchevrette) [reply pyproject.toml:62]: done in 7d409fa + fix all the problems 🎉

File: Dockerfile [hunk 5]
```diff
@@ -1,33 +1,44 @@
...
-RUN python3 -m pip install tox
-COPY --chown=$USER . $HOME
-CMD ["tox"]
-
-# prod
+#################
+# builder stage #
+#################
+FROM base AS builder
+COPY --from=ghcr.io/astral-sh/uv:0.4.29@sha256:ebb10c5178c7a357d80527f3371e7038561c26234e8a0bb323ea1f2ce8a694b7 /uv /bin/uv
```
Comment (chassing) [inline Dockerfile:12]: Please use latest uv

```suggestion
COPY --from=ghcr.io/astral-sh/uv:0.5.1@sha256:190cbcca15602bad4531b4310f928fd34a036d50ec3edb6389edc167e21c35b6 /uv /bin/uv
```

**Review by rporres — APPROVED** [review]

**Pull request merged**